types/atom/index.d.ts

```diff
@@ -305,9 +328,10 @@ export interface CommandRegistry {
 
     /**
      *  Simulate the dispatch of a command on a DOM node.
-     *  @return Whether or not there was a matching command for the target.
+     *  @return Either a Promise that resolves after all handlers complete or null if
+     *  no handlers were matched.
      */
-    dispatch(target: Node, commandName: string): boolean;
+    dispatch(target: Node, commandName: string): Promise<undefined[]> | null;
```

Comment:
This... doesn't seem to be correct either? Here's what I proposed last time around:
```ts
 export type CommandRegistryListener<TargetType extends EventTarget> = {
    didDispatch(event: CommandEvent<TargetType>): void | Promise<void>,
// ...
} | ((event: CommandEvent<TargetType>) => void | Promise<void>);
```
since `didDispatch` can apparently return a `Promise` now (although it doesn't seem like it's documented anywhere) -- at least the code handling it behaves like it expects a `Promise`.

then
```ts
dispatch(target: Node, commandName: string): Promise<void> | null
```
Why `void`? Because we can't even be sure it'll be an array, frankly: implementation might as well be `await Promise.all(matched)`,  instead of `return Promise.all(matched)`, and we shouldn't really care which one of those is it.

Does this make sense?

Reply:
`didDispatch` is definitely wrong and will be fixed. Why cant we be sure that it will be an array? `Promise.all` called with an array of zero elements will still return an array (of zero elements as well).

Reply:
`Promise.all` won't be called with zero elements anyway, since if there are zero elements in the first place, `dispatch` will return `null`. But that's beside the point.

My point is, the fact that `dispatch` returns a `Promise<Array>` seems to be an implementation detail -- since it's not documented and there doesn't seem to be any legitimate use for the value inside (?). Hence, I would propose that we might as well treat it *as if* it was `Promise<void>`, since the only meaningful thing to do with it is to await on it -- the value itself doesn't seem to be useful, so might as well pretend that there isn't any.

Does that make sense? I'm entirely open to discussion, by the way -- just I have no idea how `Promise<undefined[]>` is anything but confusing.

Reply:
Fair enough. As I mentioned up above, it was primarily so people could just check to see how many commands were executed on the `dispatch`, not really operate on the values themselves. It's not a big deal either way, but if it's potentially confusing then it can just be removed.

Reply:
Ah, I missed that bit about 
> Someone may for some reason want to know how many commands were waited on

To be honest, I really don't see much value in that. I mean, in tests one would likely have spies for that anyway, and spies will offer *much* more control.

... besides, if this was the intended use, returning `null` when there are no matches would make little sense?